modules/osdk-aws-sts-enabling.adoc

```diff
@@ -0,0 +1,319 @@
+
+[NOTE]
+====
+Pods related to the Operator deployment mount a `serviceAccountToken` volume for the secret by default.
+====
+
+.Prerequisities
+
+* {product-title} 4.14 or later
+* Cluster in STS mode
+* OLM-based Operator project
+
+.Procedure
+
+. Update your Operator project's `ClusterServiceVersion` (CSV) object:
+
+.. Ensure your Operator has RBAC permission to create `CredentialsRequests` objects:
++
+.Example `clusterPermissions` list
+[%collapsible]
+====
+[source,yaml]
+----
+# ...
+install:
+   spec:
+     clusterPermissions:
+     - rules:
+       - apiGroups:
+         - "cloudcredential.openshift.io"
+         resources:
+         - credentialsrequests
+         verbs:
+         - create
+         - delete
+         - get
+         - list
+         - patch
+         - update
+         - watch
+----
+====
+
+.. Add the following annotation to claim support for STS:
++
+[source,yaml]
+----
+# ...
+metadata:
+ annotations:
+   features.operators.openshift.io/token-auth-aws: "true"
+----
+
+. Update your Operator project code:
+
+.. Get the role ARN from the environment variable set on the pod by the `Subscription` object. For example:
++
+[source,go]
+----
+// Get ENV var
+roleARN := os.Getenv("ROLEARN")
+setupLog.Info("getting role ARN", "role ARN = ", roleARN)
+webIdentityTokenPath := "/var/run/secrets/openshift/serviceaccount/token"
+----
+
+.. Ensure you have a `CredentialsRequest` object ready to be patched and applied. For example:
++
+.Example `CredentialsRequest` object creation
+[%collapsible]
+====
+[source,go]
+----
+import (
+   minterv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+   corev1 "k8s.io/api/core/v1"
+   metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var in = minterv1.AWSProviderSpec{
+   StatementEntries: []minterv1.StatementEntry{
+       {
+           Action: []string{
+               "s3:*",
+           },
+           Effect:   "Allow",
+           Resource: "arn:aws:s3:*:*:*",
+       },
+   },
+}
+
+var codec, _ = minterv1.NewCodec()
+var ProviderSpec, _ = codec.EncodeProviderSpec(in.DeepCopyObject())
+
+const (
+   name      = "<credential_request_name>"
+   namespace = "<namespace_name>"
+)
+
+var CredentialsRequestTemplate = &minterv1.CredentialsRequest{
+   ObjectMeta: metav1.ObjectMeta{
+       Name:      name,
+       Namespace: "openshift-cloud-credential-operator",
+   },
+   Spec: minterv1.CredentialsRequestSpec{
+       ProviderSpec: ProviderSpec,
+       SecretRef: corev1.ObjectReference{
+           Name:      "<secret_name>",
+           Namespace: namespace,
+       },
+       ServiceAccountNames: []string{
+           "<service_account_name>",
+       },
+       CloudTokenString: "",
+       CloudTokenPath:   "",
+   },
+}
+----
+====
++
+Alternatively, if you are starting from a `CredentialsRequest` object in YAML form (for example, as part of the Operator bundle), you can handle it differently:
```

Comment:
adding a credReq to the bundle is not supported today

Reply:
Changed to:
```
Alternatively, if you are starting from a `CredentialsRequest` object in YAML form (for example, as part of your Operator project code), you can handle it differently:
<code_example>
+
[NOTE]
====
Adding a `CredentialsRequest` object to the Operator bundle is not currently supported.
====
```

Also took out the following comment from the code example:
```
// This YAML file might, for instance, be part of the Operator bundle
```